deploy/buildtools/buildafi.py

```diff
@@ -67,51 +97,112 @@ def aws_build(global_build_config, bypass=False):
         buildconfig.terminate_build_instance(buildconfig)
         return
 
-    rootLogger.info("Running process to build AGFI from verilog.")
-
-    # First, Produce dcp/tar for design. Runs on remote machines, out of
-    # /home/centos/firesim-build/ """
+    # First, Produce dcp/tar for design. Runs on remote machines at same path as manager
     ddir = get_deploy_dir()
+    dparent = abspath(ddir + "/..")
+    aws_fpga_root = pjoin(dparent, "platforms/f1/aws-fpga")
     buildconfig = global_build_config.get_build_by_ip(env.host_string)
-    builddir = buildconfig.get_build_dir_name()
-    # local AWS build directory; might have config-specific changes to fpga flow
+    results_builddir = buildconfig.get_build_dir_name()
+    # AWS build directory
     fpgabuilddir = "hdk/cl/developer_designs/cl_" + buildconfig.get_chisel_triplet()
-    remotefpgabuilddir = "hdk/cl/developer_designs/cl_firesim"
+    fpgatemplatedir = 'hdk/cl/developer_designs/cl_firesim'
+    generated_dir = get_sim_makevar(buildconfig, 'GENERATED_DIR')
+    fat_jar = get_sim_makevar(buildconfig, 'FAT_JAR')
 
     # first, copy aws-fpga to the build instance. it will live in
-    # firesim-build/platforms/f1/
+    # the same path on the build host as on the manager
     with StreamLogger('stdout'), StreamLogger('stderr'):
-        run('mkdir -p /home/centos/firesim-build/platforms/f1/')
+        run('mkdir -p ' + ddir)
+        run('mkdir -p ' + generated_dir)
+        run('mkdir -p ' + dirname(fat_jar))
+        run('mkdir -p ' + aws_fpga_root)
+        run('mkdir -p ' + pjoin(aws_fpga_root, fpgatemplatedir))
+        run('mkdir -p ' + pjoin(aws_fpga_root, fpgabuilddir))
+
     # do the rsync, but ignore any checkpoints that might exist on this machine
     # (in case builds were run locally)
     # extra_opts -l preserves symlinks
     with StreamLogger('stdout'), StreamLogger('stderr'):
-        rsync_cap = rsync_project(local_dir=ddir + "/../platforms/f1/aws-fpga",
-                      remote_dir='/home/centos/firesim-build/platforms/f1/',
+        rsync_cap = rsync_project(local_dir=pjoin(aws_fpga_root,''),
+                      remote_dir=aws_fpga_root,
                       ssh_opts="-o StrictHostKeyChecking=no",
                       exclude="hdk/cl/developer_designs/cl_*",
                       extra_opts="-l", capture=True)
         rootLogger.debug(rsync_cap)
         rootLogger.debug(rsync_cap.stderr)
-        rsync_cap = rsync_project(local_dir=ddir + "/../platforms/f1/aws-fpga/{}/*".format(fpgabuilddir),
-                      remote_dir='/home/centos/firesim-build/platforms/f1/aws-fpga/' + remotefpgabuilddir,
-                      exclude='build/checkpoints',
-                      ssh_opts="-o StrictHostKeyChecking=no",
-                      extra_opts="-l", capture=True)
-        rootLogger.debug(rsync_cap)
-        rootLogger.debug(rsync_cap.stderr)
+
+    if global_build_config.distributed_elaboration:
```

Comment:
@davidbiancolin good call having me put this behind an option.  I'm using this because I need to manually hack my way round the DCE issue for now.